code/modules/mob/living/carbon/human/human.dm

```diff
@@ -4,14 +4,104 @@
 	voice_name = "Unknown"
 	icon = 'icons/mob/human.dmi'
 	icon_state = "caucasian1_m_s"
-
-
+	var/infected = 0
+	var/infection = 0
+	var/startinfected = 1
+	var/zombification = 0
+	var/numinfectedh = 0
 
 /mob/living/carbon/human/dummy
 	real_name = "Test Dummy"
 	status_flags = GODMODE|CANPUSH
 
-
+/mob/living/carbon/human/proc/oldInfect(mob/living/carbon/human/H)
+	if(H.startinfected == 1 && H.infection == 0)
+		H.infected = 1
+		H.infection = 1
+		H.faction = list("zombie")
+		H << "You feel slightly ill..."
+		spawn(rand(700, 800))
+			H << "<span class='userdanger'>Something really is not right..</span>"
+			visible_message("<b>[H]</b> looks very pale...")
+			H.adjustOxyLoss(20)
+			H.stuttering = 10
+			spawn(rand(800, 900))
+				H << "<span class='userdanger'>You feel like you could die at any moment...</span>"
+				visible_message("<b>[H]</b> begins sweating uncontrollably!")
+				H.adjustOxyLoss(60)
+				H.stuttering = 20
+				H.Weaken(10)
+				H.Stun(5)
+				spawn(rand(800, 900))
+					H << "<span class='userdanger'>You are about to pass out!</span>"
+					visible_message("<b>[H]</b> begins bleeding uncontrollably!")
+					H.adjustOxyLoss(70)
+					H.adjustBloodLoss(20)
+					spawn(rand(800, 900))
+						H << "<span class='userdanger'>Please just end the pain!</span>"
+						visible_message("<b>[H]</b> begins making crunching noises, their skin looks almost blue!")
+						H.adjustBloodLoss(50)
+						H.adjustBruteLoss(40)
+						H.Weaken(10)
+						H.stuttering = 10
+						H.Stun(5)
+						oldZombify(H)
+						H.startinfected = 0
+						H.infection = 0
+	else if(H.infection == 0)
+		oldZombify(H)
+		H.infection = 1
+
+/mob/living/carbon/human/proc/oldZombify(mob/living/carbon/human/H)
+	if(zombification == 0)
+		H.zombification = 1
+		visible_message("<span class='danger'>[H] looks a bit odd.. their skin is basically blue..</span>")
+		H << "<span class='userdanger'>You don't feel right! Something is wrong!</span>"
+		H.faction = list("zombie")
+		spawn(rand(100, 200))
+			H.stat = DEAD
+			spawn(rand(200,300))
+				H.set_species(/datum/species/zombie)
+				if(H.head) //So people can see they're a zombie
+					var/obj/item/clothing/helmet = H.head
+					if(!H.unEquip(helmet))
+						qdel(helmet)
+				if(H.wear_mask)
+					var/obj/item/clothing/mask = H.wear_mask
+					if(!H.unEquip(mask))
+						qdel(mask)
+				var/mob/living/simple_animal/hostile/oldzombie/Z = new /mob/living/simple_animal/hostile/oldzombie(H.loc)
+				Z.faction = src.faction
+				Z.appearance = H.appearance
+				Z.transform = matrix()
+				Z.pixel_y = 0
+				for(var/mob/dead/observer/ghost in player_list)
+					if(H.real_name == ghost.real_name)
+						ghost.reenter_corpse()
```

Comment:
It might be nice if the ghost in question is prompted if he/she wants to play a zombie, otherwise allow other ghosts to possess?